### ISSUE TITLE
Balanced accuracy comparison with sk-learn

### ISSUE THREAD

**Issue by ablaom** [JuliaAI/MLJBase.jl#651]
In [MLJBase tests](https://github.com/JuliaAI/MLJBase.jl/blob/28dec9db035a69112a64b5b772454f8162387f85/test/measures/finite.jl#L38) (which pass) there is a score for the sk-learn implementation used for comparison that I cannot reproduce:

```python
In [8]: from sklearn.metrics import balanced_accuracy_score

In [9]: y = [3, 4, 1, 1, 1, 4, 1, 3, 3, 1, 2, 3, 1, 3, 3, 3, 2, 4, 3, 2, 1, 3,
   ...:     3, 1, 1, 1, 2, 4, 1, 4, 4, 4, 1, 1, 4, 4, 3, 1, 2, 2, 3, 4, 2, 1,
   ...:     2, 2, 3, 2, 2, 3, 1, 2, 3, 4, 1, 2, 4, 2, 1, 4, 3, 2, 3, 3, 3, 1,
   ...:     3, 1, 4, 3, 1, 2, 3, 1, 2, 2, 4, 4, 1, 3, 2, 1, 4, 3, 3, 1, 3, 1,
   ...:     2, 2, 2, 2, 2, 3, 2, 1, 1, 4, 2, 2]

In [10]: ŷ = [2, 3, 2, 1, 2, 2, 3, 3, 2, 4, 2, 3, 2, 4, 3, 4, 4, 2, 1, 3, 3, 3,
   ...:     3, 3, 2, 4, 4, 3, 4, 4, 1, 2, 3, 2, 4, 1, 2, 3, 1, 4, 2, 2, 1, 2,
   ...:     3, 2, 2, 4, 3, 2, 2, 2, 1, 2, 2, 1, 3, 1, 4, 1, 2, 1, 2, 4, 3, 2,
   ...:     4, 3, 2, 4, 4, 2, 4, 3, 2, 3, 1, 2, 1, 2, 1, 2, 3, 1, 1, 3, 4, 2,
   ...:     4, 4, 2, 1, 3, 2, 2, 4, 1, 1, 4, 1]

In [11]: balanced_accuracy_score(ŷ, y)

Out[11]: 0.17544466403162054
```

In tests the claim is for `0.17493386243386244` which is ball-park close but not *that* close.

I got the same answer using a newer and mach older python/sklearn installation (2.7.13 / 0.18.1)

@tlienart Are you still able to reproduce this python result, or recall anything to explain the difference here?

**Comment by ablaom** [comment]
The same issue exists for the weighted version, and (after an imminent PR), the adjusted version.

**Comment by tlienart** [comment]
I think this should clarify it:

```
>>> balanced_accuracy_score(ŷ, y)
0.17544466403162054
>>> balanced_accuracy_score(y, ŷ)
0.17493386243386244
```

the standard in sklearn is to put the ground truth first (see: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.balanced_accuracy_score.html)

**Comment by ablaom** [comment]
Okay, I feel pretty stupid 🙄 . Thanks for this prompt resolution!

**Comment by tlienart** [comment]
Haha no worries, glad I could help in this case!
